cpp/test/sg/rf_treelite_test.cu

```diff
@@ -440,19 +438,19 @@ INSTANTIATE_TEST_CASE_P(RfBinaryClassifierTreeliteTests,
                         ::testing::ValuesIn(inputsf2_clf));
 
 const std::vector<RfInputs<float>> inputsf2_reg = {
-  {4, 2, 1, 1.0f, 1.0f, 4, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0,
-   CRITERION::MSE, 1},
-  {40, 20, 1, 1.0f, 1.0f, 40, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0,
-   CRITERION::MSE, 1},
-  {40, 20, 5, 1.0f, 1.0f, 40, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0,
-   CRITERION::CRITERION_END,
-   1},  // 0.0, CRITERION_END uses the default criterion (GINI for classification, MSE for regression)
-  {40, 20, 1, 1.0f, 1.0f, 40, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0,
-   CRITERION::MAE, 1},
-  {400, 200, 1, 1.0f, 1.0f, 400, 8, -1, false, false, 4,
-   SPLIT_ALGO::GLOBAL_QUANTILE, 2, 0.0, CRITERION::MAE, 1},
-  {400, 200, 5, 1.0f, 1.0f, 400, 8, -1, true, false, 4, SPLIT_ALGO::HIST, 2,
-   0.0, CRITERION::CRITERION_END, 1}};
+  {4, 2, 1, 1.0f, 1.0f, 4, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0, 2,
+   CRITERION::MSE},
+  {4, 2, 1, 1.0f, 1.0f, 4, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0, 2,
+   CRITERION::MSE},
+  {4, 2, 5, 1.0f, 1.0f, 4, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0, 2,
+   CRITERION::
+     CRITERION_END},  // CRITERION_END uses the default criterion (GINI for classification, MSE for regression)
+  {4, 2, 1, 1.0f, 1.0f, 4, 8, -1, false, false, 4, SPLIT_ALGO::HIST, 2, 0.0, 2,
+   CRITERION::MAE},
+  {4, 2, 1, 1.0f, 1.0f, 4, 8, -1, false, false, 4, SPLIT_ALGO::GLOBAL_QUANTILE,
+   2, 0.0, 2, CRITERION::MAE},
+  {4, 2, 5, 1.0f, 1.0f, 4, 8, -1, true, false, 4, SPLIT_ALGO::HIST, 2, 0.0, 2,
```

Comment:
Do any of these tests grow to a level that would really stress the old algorithm memory-wise and require the gather approach? I think they are relatively small, right? We could also add one as a "stress" test (not run every time but only on demand) at the python layer if the good stress tests are too slow.

Reply:
I'd prefer if we do such kind of stress tests at the python layer itself.

Reply:
I'm ok with that. I filed Issue #1552 as a reminder.